### PR TITLE
pipeline: don't re-add active tasks and copy immediately

### PR DESCRIPTION
**Pull request by lyakh**
In a topology with two connected pipelines, e.g. two playback pipelines, connected on a mixer, if one PCM is already streaming and then the other one is started too, two bugs trigger:
1. when adding pipelines to the scheduler, all pipelines are added, including any common pipelines behind the mixer
2. the mixer pipeline then gets to run first, while it has no data available to it yet.

To fix these bugs
1. don't re-add any active pipelines
2. when a pipeline is added for streaming, let it begin copying immediately instead of skipping a period.

### CODE REVIEW

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lyakh** [comment]
No changes, force CI re-test

File: src/audio/pipeline/pipeline-schedule.c [hunk 1]
```diff
@@ -370,6 +370,9 @@ void pipeline_schedule_copy(struct pipeline *p, uint64_t start)
 	if (!pipeline_is_timer_driven(p))
 		sa_set_panic_on_delay(false);
 
+	if (task_is_active(p->pipe_task))
```
Comment (lgirdwood) [inline src/audio/pipeline/pipeline-schedule.c:373]: Can we have a comment saying why we do this check.

Comment (lyakh) [reply src/audio/pipeline/pipeline-schedule.c:380]: @lgirdwood done. BTW, this also *might* help with some IPC timeouts, currently in some cases the DAI pipeline can be scheduled before the host pipeline for playback, in which case of course the DAI pipeline has to skip a scheduling tick.

**Comment by lyakh** [comment]
CI: 3 devices were not available (CML, WHL and GLK) and two compilations failed because of licence server inavailability

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lyakh** [comment]
CI: a single failure of a suspend-resume test on ADL: https://sof-ci.01.org/sofpr/PR6165/build1319/devicetest/?model=ADLP_RVP_NOCODEC&testcase=check-suspend-resume-with-capture-5

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**